pkg/hooks/loader.go

```diff
@@ -198,10 +198,18 @@ func (h *Hook) GetTcsMocks() ([]*models.Mock, error) {
 	return tcsMocks, nil
 }
 
-func (h *Hook) IsUsrAppTerminateInitiated() bool {
+func (h *Hook) IsUserAppTerminateInitiated() bool {
+	h.mutex.Lock()
```

Comment:
Try to use to different mutex object. Same mutex is used for appending mocks. This mutex may collide with appending mocks mutex.